### PR TITLE
Add Sieve Of Eratosthenes in TypeScript

### LINKED ISSUE #2174: Sieve Of Eratosthenes
- [x] C
- [x] C++
- [ ] C# - @kashika3101
- [x] CoffeeScript
- [ ] Dart - @shivang2k
- [x] Go
- [x] Java
- [x] JS
- [x] Kotlin
- [x] Php - @mruna7
- [x] Python
- [x] Readme
- [ ] Ruby - @Basilleaf24
- [ ] Rust
- [x] TypeScript - @shivamgupta57121

### PR DESCRIPTION
**Pull request by shivamgupta57121**
Fixes #2174

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Add code for Sieve Of Eratosthenes in TypeScript

#### Languages Used:
- TypeScript

#### Files Added:
- Sieve_Of_Eratosthenes.ts

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

File: Sieve_Of_Eratosthenes/Sieve_Of_Eratosthenes.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+    // Create a boolean array "prime[0..n]" and initialize
+    // all entries of it as true. A value in prime[i] will
+    // finally be false if i is Not a prime, else true.
+
+    let prime : boolean[] = [true];
+    let p : number;
+    let i : number;
+
+    for(p=0; p<=n; p++)
```
Comment (asha15) [inline Sieve_Of_Eratosthenes/Sieve_Of_Eratosthenes.ts:13]: please keep space around operators

File: Sieve_Of_Eratosthenes/Sieve_Of_Eratosthenes.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        prime.push(true);
+
+    for(p = 2; p*p <= n; p++){
```
Comment (asha15) [inline Sieve_Of_Eratosthenes/Sieve_Of_Eratosthenes.ts:16]: keep space around operator

File: Sieve_Of_Eratosthenes/Sieve_Of_Eratosthenes.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        // If prime[p] is true, then it is a prime
+        if(prime[p]){
+            // Update all multiples of p
+            for(i = p*p; i <= n; i += p)
```
Comment (asha15) [inline Sieve_Of_Eratosthenes/Sieve_Of_Eratosthenes.ts:20]: please keep space around operators

File: Sieve_Of_Eratosthenes/Sieve_Of_Eratosthenes.ts (hunk 2 above)
Comment (shivamgupta57121) [reply Sieve_Of_Eratosthenes/Sieve_Of_Eratosthenes.ts]: Added

File: Sieve_Of_Eratosthenes/Sieve_Of_Eratosthenes.ts (hunk 3 above)
Comment (shivamgupta57121) [reply Sieve_Of_Eratosthenes/Sieve_Of_Eratosthenes.ts]: Added

**Comment by shivamgupta57121** [comment]
@asha15 Added the space around operators. Please review it.

File: Sieve_Of_Eratosthenes/Sieve_Of_Eratosthenes.ts (hunk 1 above)
Comment (shivamgupta57121) [reply Sieve_Of_Eratosthenes/Sieve_Of_Eratosthenes.ts]: Added

**Comment by raghav-dalmia** [comment]
Looks good to me. Just squash your commits, I will merge it after that.

**Comment by shivamgupta57121** [comment]
Squashed the changes. Please merge @raghav-dalmia

**Pull request merged**